storm-client/src/jvm/org/apache/storm/stats/BoltExecutorStats.java

```diff
@@ -53,10 +58,53 @@ public void cleanupStats() {
         super.cleanupStats();
     }
 
-    public void boltExecuteTuple(String component, String stream, long latencyMs) {
+    public void boltExecuteTuple(String component, String stream, long latencyMs, long workerUptimeSecs,
+                                 Task firstExecutorTask) {
         List key = Lists.newArrayList(component, stream);
         this.getExecuted().incBy(key, this.rate);
         this.getExecuteLatencies().record(key, latencyMs);
+
+        // Calculate capacity:  This is really for the whole executor, but we will use the executor's first task
+        // for reporting the metric.
+        double capacity = calculateCapacity(workerUptimeSecs);
+        firstExecutorTask.getTaskMetrics().setCapacity(capacity);
+    }
+
+    private double calculateCapacity(long workerUptimeSecs) {
+        if (workerUptimeSecs > 0) {
+            Map<String, Double> execAvg = valueStat(this.getExecuteLatencies()).get(MultiCountStat.TEN_MIN_IN_SECONDS_STR);
```

Comment:
`TEN_MIN_IN_SECONDS_STR` can be a local variable.  But this is not important.

Reply:
I was going to use MultiCountStat.TEN_MIN_IN_SECONDS_STR to replace the one in StatsUtil, which is in storm-server for more general usage, but didn't replace the other usages to keep a smaller review.

Reply:
There is another similar class `MultiLatencyStat`. It doesn't feel right to use `MultiCountStat.TEN_MIN_IN_SECONDS_STR` for `MultiLatencyStat`. But this is just personal preference and we can leave this to the future. I am okay with current change.